calendar-bundle/tests/EventListener/BreadcrumbListenerTest.php

```diff
@@ -0,0 +1,384 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CalendarBundle\Tests\EventListener;
+
+use Contao\CalendarEventsModel;
+use Contao\CalendarModel;
+use Contao\Config;
+use Contao\CoreBundle\Framework\Adapter;
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Contao\Events;
+use Contao\Input;
+use Contao\CalendarBundle\EventListener\BreadcrumbListener;
+use Contao\PageModel;
+use Contao\TestCase\ContaoTestCase;
+use PHPUnit\Framework\MockObject\MockObject;
+
+final class BreadcrumbListenerTest extends ContaoTestCase
```

Comment:
Please do not make classes final unless absolutely necessary.

Reply:
Any reasons why you prefer keeping unit tests and infrastructure classes (listeners) extenable? Usually the advice is do not make class not final unless absolutely necessary. ;) However, removed in bbe1d28.